hardkernel/odroid-m1/default.nix

```diff
@@ -0,0 +1,33 @@
+# Based on kboot-conf by original authors povik and sstent.
+# I'm just porting things over. the original work can be viewed at:
+# https://github.com/sstent/nixos-on-odroid-m1 
+# https://github.com/povik/nixos-on-odroid-n2
+{ config, pkgs, lib, ... }:
+
+{
+  imports = [
+    ./kboot-conf
+  ];
+
+  #boot.loader.grub.enable = false;
+  boot.loader.kboot-conf.enable = true;
+  # Use kernel >6.6 The devicetree is missing from kernel versions older than this.
+  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "6.6") (lib.mkDefault pkgs.linuxPackages_latest);
+  # Stop ZFS breaking the build
+  boot.supportedFilesystems = lib.mkForce ["btrfs" "cifs" "f2fs" "jfs" "ntfs" "reiserfs" "vfat" "xfs"];
```

Comment:
This can be potentially dangerous if other module try to add something. Why is it needed for `linuxPackages_latest`? Nothing should add zfs by default.

Reply:
I don't think it'd be necessary for `linuxPackages_latest`, I'll do some tests and remove `boot.supportedFilesystems`.

Reply:
Removing the `boot.supportedFilesystems` broke the boot flow, I haven't figured out why yet.